chalk-integration/src/lowering.rs

```diff
@@ -1031,6 +1031,7 @@ impl LowerFnDefn for FnDefn {
 
         Ok(rust_ir::FnDefDatum {
             id: fn_def_id,
+            abi: ChalkFnAbi::Rust,
```

Comment:
Locally I have added a `#[fn_abi(Rust/C)]` annotation for functions to control the ABI, but I haven't committed it because I don't know if we can use it in the tests currently (I don't think you can equate function definitions?), or if it's even desired.

Reply:
hmm, yeah, it seems like we need a way to test this, so I think you should add it, though I'd probably prefer 

```
extern "C" fn() { ... }
```

notation to `#[fn_abi(C)]`. You can handle that by parsing a string literal and comparing it against known values...